packages/manager/apps/pci-kubernetes/src/api/hooks/useKubernetes.ts

```diff
@@ -39,10 +39,22 @@ export const getAllKubeQueryKey = (projectId: string) => [
   'kube',
 ];
 
-export const useAllKube = (projectId: string) =>
+export const useAllKube = (projectId: string, refetchIntervalTime?: number) =>
   useQuery({
     queryKey: getAllKubeQueryKey(projectId),
     queryFn: (): Promise<Required<TKube[]>> => getAllKube(projectId),
+    refetchInterval(query) {
+      if (!refetchIntervalTime) {
+        return false;
+      }
+      const notAllReady = query.state.data?.some(
+        (data) => data.status !== 'READY',
+      );
+      if (notAllReady) {
+        return refetchIntervalTime;
+      }
+      return false;
+    },
```

Comment:
As the `refetchInterval` function is declared twice (here and in `useKubernetesCluster`), you can create a dedicated function that takes the predicate function as parameter 😌.

And, additionally, if you want to be compliant with inversion of dependencies principle, you can give the function as the hook parameter 😎